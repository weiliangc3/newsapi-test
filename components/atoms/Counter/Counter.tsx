import React, { FunctionComponent, useState } from 'react';

// Sample component
const Counter: FunctionComponent<Props> = ({ initialCount = 0 }: Props) => {
  const [count, setCount] = useState<number>(initialCount);
  return (
    <div>
      <span data-testid="count">{count}</span>
      <button
        onClick={() => { setCount(count + 1); }}
        type="button"
      >
        Increment
      </button>
    </div>
  );
};

export default Counter;

interface Props{
  initialCount?: number,
}
