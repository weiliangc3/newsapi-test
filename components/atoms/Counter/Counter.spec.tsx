import React from 'react';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';
import { render, screen } from '@testing-library/react';
import Counter from './Counter';

// Sample test
describe('Counter', () => {
  it('renders correctly', () => {
    const tree = render(<Counter />);
    expect(tree.container).toMatchSnapshot();
  });

  it('takes a default value', () => {
    const tree = render(<Counter initialCount={10} />);
    expect(tree.container).toMatchSnapshot();
  });

  it('increments when the button is clicked', () => {
    render(<Counter />);
    expect(screen.getByTestId('count')).toContainHTML('0');
    userEvent.click(screen.getByText('Increment'));
    expect(screen.getByTestId('count')).toContainHTML('1');

    // alternatively:
    // expect(screen.getByText('1')).toBeInTheDocument();
  });
});
