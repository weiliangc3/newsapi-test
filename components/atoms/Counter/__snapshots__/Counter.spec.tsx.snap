// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter renders correctly 1`] = `
<div>
  <div>
    <span
      data-testid="count"
    >
      0
    </span>
    <button
      type="button"
    >
      Increment
    </button>
  </div>
</div>
`;

exports[`Counter takes a default value 1`] = `
<div>
  <div>
    <span
      data-testid="count"
    >
      10
    </span>
    <button
      type="button"
    >
      Increment
    </button>
  </div>
</div>
`;
